rfcs/detecting_sensitive_textual_content/20230309-project_proposal_detecting_sensitive_textual_content.md

```diff
@@ -0,0 +1,379 @@
+# Project Proposal: Detecting sensitive textual content
+
+## Reviewers
+
+- [ ] @aetherunbound
+- [ ] @krysal
+
+## Project summary
+
+Detect results with sensitive textual content by checking their textual content
+against a list of sensitive terms. Once results with sensitive terms in their
+textual content are determined, blur those results on the frontend.
+
+## Goals
+
+This advances Openverse's goal of increasing the safety of the platform.
+
+## Requirements
+
+> **A note on tone**: This section should be mostly readable and understandable
+> by casual, non-technical readers. However, certain aspects of it unavoidably
+> delve into technical details in order to describe how features fit together or
+> considerations that will need to be kept in mind during implementation
+> planning. These need to be documented _somewhere_, and displacing them from
+> the requirements felt strange. If you are not interested in these particular
+> technical details, please skip the two sections titled "Implementation
+> details" and the section on "How to blur results". These sections in
+> particular consider technical issues of the project that may confound readers
+> unfamiliar with software development whether applied to Openverse or more
+> generally.
+
+### API
+
+#### Filtering
+
+The Django API is able to filter out results that include sensitive terms in
+their textual content. It does so without degrading search performance.
+
+Filtering of results with sensitive terms in their textual content occurs when
+`include_sensitive_results=True`. This parameter will supplant the `mature=True`
+as a more comprehensively and descriptively named parameter. `mature=True`
+should still work, but it should just be an alias for
+`include_sensitive_results` and should be marked as deprecated in or altogether
+excluded from the API documentation.
+
+##### Implementation details
+
+This will likely be implemented as a secondary index of results that do not
+include the sensitive terms, as explored in
+https://github.com/WordPress/openverse-api/pull/1108.
+
```

Comment:
Would determining the list of "sensitive terms" be a part of this project?

Reply:
Yes! I should add that as another one of the requested documents aside from the implementation plans. I believe @zackkrida is already sort of working on this: https://github.com/WordPress/openverse/issues/377#issuecomment-1458653924

I suspect that was delayed due to Zack being sick, or I just haven't seen it yet today :slightly_smiling_face:

Reply:
Yep, I'm working on this today still! Expecting to publish in an hour or two.

Reply:
Done, here: https://github.com/WordPress/openverse/pull/911